src/stream/tests/integration_tests/snapshot.rs

```diff
@@ -98,6 +99,19 @@ pub async fn check_with_script<F, Fut>(
     expect.assert_eq(&output);
 }
 
+pub async fn check_with_script_v2<F, Fut>(
+    create_executor: F,
+    test_script: &str,
+    expect: expect_test::Expect,
+    options: SnapshotOptions,
+) where
+    F: Fn() -> Fut,
+    Fut: Future<Output = (MessageSender, BoxedMessageStream)>,
+{
+    let output = executor_snapshot_v2(create_executor, test_script, options).await;
+    expect.assert_eq(&output);
+}
+
```

Comment:
Oh I see this is already introduced. Please use a more intuitive name and add some comment.